### PR TITLE
Basic auth and navigation

### PR DESCRIPTION
**Pull request by thequeenofspades**
- Adds authentication libraries 'rest_framework.authtoken,' 'dj_rest_auth,' and 'rest_framework_simplejwt' to the backend.
- Adds a login screen and a signup screen to the frontend so users can create an account and log in. Email verification is turned off for now. User accounts are stored in the local Django DB and can be modifed, created, and deleted using Django's built-in admin interface.
- Adds basic navigation to the app in the form of a sidebar with 4 links: Login/Signup, Home, Map, Logout. Login/Signup are only shown to logged out users, and Logout is only shown to logged in users.
- Doesn't include tests yet.
- Known issues: logout is buggy (doesn't always work on the first try). Homepage shows empty string when the user is logged out. I'll keep working on these, but thought it was worth merging the navigation changes early to avoid merge conflicts down the road.

### CODE REVIEW

**Review by riecke — APPROVED** [review]
We should probably add tests for components soon-ish, once we start to get the structure.

File: src/interface/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+import { LoginComponent } from './login/login.component';
+import { MapComponent } from './map/map.component';
+import { SignupComponent } from './signup/signup.component';
+
+const routes: Routes = [
+  { path: '', redirectTo: 'login', pathMatch: 'full' },
+  { path: 'login', component: LoginComponent },
+  { path: 'signup', component: SignupComponent },
+  { path: 'home' , component: HomeComponent, canActivate: [AuthGuard] },
+  { path: 'map', component: MapComponent, canActivate: [AuthGuard]},
```
Comment (lattemilk) [inline src/interface/src/app/app-routing.module.ts]: Minor nit - a space is missing after [AuthGuard[ on this line.

**Comment by lattemilk** [comment]
this is great! 👍

File: src/interface/src/app/app-routing.module.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/app-routing.module.ts]: FYI I think you're looking at an old commit. These lines are different as of the latest commit to the branch (the paths are no longer guarded by AuthGuard).

File: src/interface/src/app/app.module.ts [hunk 2]
```diff
@@ -1,30 +1,57 @@
...
 
 @NgModule({
   declarations: [
     AppComponent,
-    MapComponent
+    MapComponent,
+    LoginComponent,
+    SignupComponent,
+    NavigationComponent,
+    HomeComponent
```
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/app.module.ts:33]: Yep, I have no problem with that. Will do!

File: src/interface/src/app/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+
+  loggedInStatus$: Subject<boolean> = new Subject();
+  isLoggedIn$: Observable<boolean> = this.loggedInStatus$.pipe(shareReplay(1));
+
+  private apiRoot = 'http://localhost:8000/dj-rest-auth/';
```
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/auth.service.ts:15]: Done

File: src/interface/src/app/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+  constructor(
+    private http: HttpClient,
```
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/auth.service.ts:18]: Done

File: src/interface/src/app/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    private cookieService: CookieService,
+    private router: Router) {
+      this.isLoggedIn$.subscribe();
+      if (this.isLoggedIn()) {
```
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/auth.service.ts:22]: done!

File: src/interface/src/app/login/login.component.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+<form (ngSubmit)="onSubmit()">
+  <label for="username">Username</label>
+  <input  type="text" id="username"
+          required
+          [(ngModel)]="model.username" name="username"
+          #name="ngModel">
+
+  <label for="password">Password</label>
+  <input  type="password" id="password"
```
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/login/login.component.html:15]: Intentional to align with the properties on the lines below. Let me know if it's against style practices; I saw it done this way in an Angular tutorial so assumed it was standard.

File: src/interface/src/app/login/login.component.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<div style="text-align:center">
+  <h1>
+    Login
+  </h1>
+</div>
+
+<form (ngSubmit)="onSubmit()">
+  <label for="username">Username</label>
+  <input  type="text" id="username"
```
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/login/login.component.html:9]: same reason as above

File: src/interface/src/app/login/login.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (lattemilk) [reply src/interface/src/app/login/login.component.html:15]: I see. Hmm I haven't seen it formatted like that (or maybe I never noticed lol) Can these be aligned without the extra space in front of "type"? Or you can move type="password" to the next line and have it indented?

**Review by lattemilk — APPROVED** [review]

File: src/interface/src/app/login/login.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (thequeenofspades) [reply src/interface/src/app/login/login.component.html]: Sure, done.

**Pull request merged**